flink-streaming-java/src/test/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironmentComplexConfigurationTest.java

```diff
@@ -72,7 +73,7 @@ public void testLoadingStateBackendFromConfiguration() {
                 configuration, Thread.currentThread().getContextClassLoader());
 
         StateBackend actualStateBackend = envFromConfiguration.getStateBackend();
-        assertThat(actualStateBackend, instanceOf(MemoryStateBackend.class));
+        assertThat(actualStateBackend, nullValue());
```

Comment:
This test becomes nonsense if changing it like this. I would remove it as long as there is a test to cover the case that the state backend set via `StreamExecutionEnvironment#configure` can be properly created at runtime.

Reply:
Since there are already corresponding unit tests covering the creation of StateBackend, CheckpointStorage, and Restart Strategy from the job configuration of the execution graph, I have added a case, testJobConfigFromEnvToExecutionGraph, to cover the transfer of the configuration from the environment to the execution graph as the job configuration.